### ISSUE TITLE
Client-side routing should be screen reader-accessible

### ISSUE THREAD

**Issue by jhackett1** [gatsbyjs/gatsby#5533]
## Summary

At the moment, screen readers don't announce when a client side router changes the page.

### Basic example

If you click on a Gatsby-Link element with a screen reader turned on (Voiceover with Chrome in my case), nothing is announced and users have no way of knowing that the page has changed.

### Motivation

This is a common issue with client-side routers, but there are [documented solutions](https://medium.com/@robdel12/single-page-apps-routers-are-broken-255daa310cf).

Would it be possible to incorporate a solution into Gatsby by default?

It would set a really great example if Gatsby made sure that generated sites are usable by everyone, and it would be especially appreciated by developers working on projects where we're legally obliged to make sure our work is accessible.

**Comment by pieh** [comment]
I'll close this as duplicate of #5581 (which is newer but has more information and some discussion already going on)
